contrib/format-maprdb/src/test/java/com/mapr/drill/maprdb/tests/index/IndexPlanTest.java

```diff
@@ -0,0 +1,1715 @@
+        .build()
+        .run();
+  }
+
+  @Test //negative case for no filter plan
+  public void testNoFilterOrderByNoIndexMatch() throws Exception {
+    String query = "SELECT t.`id`.`ssn` AS `ssn`, t.contact.phone as phone FROM hbase.`index_test_primary` as t " +
+        "order by t.name.fname limit 2";
+    test(defaultHavingIndexPlan);
+    PlanTestBase.testPlanMatchingPatterns(query,
+        new String[] {"(Sort|TopN)"},
+        new String[]{"indexName="}
+    );
+  }
+
+// Enable this testcase once MD-2848 is fixed.
+//  @Test
+//  public void IntersectPlanWithOneSideNoRows() throws Exception {
+//    try {
+//      String query = "SELECT t.`name`.`lname` AS `lname` FROM hbase.`index_test_primary` as t " +
+//              " where t.personal.age = 53 AND t.personal.income=111145";
+//      test(defaultHavingIndexPlan);
+//      test(preferIntersectPlans + ";" + disableFTS);
+//      PlanTestBase.testPlanMatchingPatterns(query,
+//              new String[]{"RowKeyJoin(.*[\n\r])+.*RestrictedJsonTableGroupScan(.*[\n\r])+.*HashJoin(.*[\n\r])+.*JsonTableGroupScan.*indexName=(i_age|i_income)(.*[\n\r])+.*JsonTableGroupScan.*indexName=(i_age|i_income)"},
+//              new String[]{}
+//      );
+//
+//      testNoResult(query);
+//
+//    } finally {
+//      test(defaultIntersectPlans + ";" + enableFTS);
+//    }
+//  }
+
+  //"i_cast_age_state_phone", "$CAST(personal.age@STRING),address.state,contact.phone", "name.fname",
+  @Test
+  public void testTrailingFieldIndexCovering() throws Exception {
+    String query = "SELECT t.`name`.`fname` AS `fname` FROM hbase.`index_test_primary` as t " +
+        " where cast(t.personal.age as INT)=53 AND t.contact.phone='6500005471' ";
+
+    test(defaultHavingIndexPlan);
+    PlanTestBase.testPlanMatchingPatterns(query,
+        new String[] {"indexName=i_cast_age_income_phone"},
+        new String[]{"RowKeyJoin"}
+    );
+
+    testBuilder()
+        .ordered()
+        .sqlQuery(query)
+        .baselineColumns("fname").baselineValues("KfFzK")
+        .build()
+        .run();
+  }
+
+  @Test
+  public void testIncludedFieldCovering() throws Exception {
+    String query = "SELECT t.`contact`.`phone` AS `phone` FROM hbase.`index_test_primary` as t " +
+        " where cast(t.personal.age as INT)=53 AND t.name.fname='KfFzK' ";
+
+    test(defaultHavingIndexPlan);
+    PlanTestBase.testPlanMatchingPatterns(query,
+        new String[] {"indexName=i_cast_age_income_phone"},
+        new String[]{"RowKeyJoin"}
+    );
+
+    testBuilder()
+        .ordered()
+        .sqlQuery(query)
+        .baselineColumns("phone").baselineValues("6500005471")
+        .build()
+        .run();
+  }
+
+  @Test
+  public void testWithFilterGroupBy() throws Exception {
+    String query = " select t1.driverlicense from hbase.`index_test_primary` t1" +
+            " where t1.driverlicense > 100000001 group by t1.driverlicense limit 2";
+    try {
+      test(defaultHavingIndexPlan);
+      test(disableHashAgg);
```

Comment:
@vdiravka since this is a fairly large test file (close to 75 tests) this suggestion and couple of other suggestions for the test changes seem nice-to-have but perhaps not must-have ?  In the interest of getting the main code changes merged, would it be okay to defer this ?

Reply:
Surely, there are a lot of places in Drill unit tests, which can be refactored. At least `BaseTestQuery` is marked as `@Deprecated`, but is widely used.
Possibly we can gather all this small issues and make the global refactoring in the process of some task, for example in the process of updating JUnit to 5 version.